src/client/pfs/pfs.proto

```diff
@@ -79,6 +64,39 @@ message RepoAuthInfo {
   auth.Scope access_level = 1;
 }
 
+message BranchInfo {
+  Branch branch = 4;
+  Commit head = 2;
+  repeated Branch provenance = 3;
+  repeated Branch subvenance = 5;
+  repeated Branch direct_provenance = 6;
+  Trigger trigger = 7;
+
+  // Deprecated field left for backward compatibility.
+  string name = 1;
+}
+
+message BranchInfos {
+  repeated BranchInfo branch_info = 1;
+}
+
+// Trigger defines the conditions under which a head is moved, and to which
+// branch it is moved.
+message Trigger {
+  // Which branch this trigger refers to
+  string branch = 1;
+  // All indicates that all conditions must be satisfied before the trigger
+  // happens, otherwise any conditions being satisfied will trigger it.
+  bool all = 2;
+  // Triggers if the cron spec has been satisfied since the last trigger and
+  // there's been a new commit.
+  string cron_spec = 3;
+  // Triggers if there's been `size` new data added since the last trigger.
+  string size = 4;
```

Comment:
It's a string to match with the request strings in the pipeline spec since that seemed like the best analogy. Also because people who are calling it through the golang or python api should be able to use the same syntax as with pachctl so it makes sense to parse the string server side.

Reply:
In terms of the name, `NumBytes` didn't seem quite right because the string actually specifies the units whether it's bytes or gigabytes. I guess it's all bytes at the end of the day, we have other fields that are called `SizeBytes` but those are int64s and always just represent the number of bytes in a dumb way. When I thought about this it didn't seem like there was anything better than `Size` but I didn't think too hard about it, I'm definitely open to suggestions though, I didn't spend too much time thinking about alternatives.